bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoInsightHandler.java

```diff
@@ -71,94 +64,21 @@ public void onValueReceived(@Nullable String variable, @Nullable String value, @
-                        energyTodayMWMin.divide(new BigDecimal(60000), 0, RoundingMode.HALF_UP), Units.WATT_HOUR);
-                logger.trace("New InsightParam energyToday '{}' for device '{}' received", energyToday,
-                        getThing().getUID());
-                updateState(WemoBindingConstants.CHANNEL_ENERGYTODAY, energyToday);
-
-                BigDecimal energyTotalMWMin = new BigDecimal(splitInsightParams[9]);
-                // recalculate mW-mins to Wh
-                State energyTotal = new QuantityType<>(
-                        energyTotalMWMin.divide(new BigDecimal(60000), 0, RoundingMode.HALF_UP), Units.WATT_HOUR);
-                logger.trace("New InsightParam energyTotal '{}' for device '{}' received", energyTotal,
-                        getThing().getUID());
-                updateState(WemoBindingConstants.CHANNEL_ENERGYTOTAL, energyTotal);
-
-                if (splitInsightParams.length > 10 && splitInsightParams[10] != null) {
-                    BigDecimal standByLimitMW = new BigDecimal(splitInsightParams[10]);
-                    State standByLimit = new QuantityType<>(
-                            standByLimitMW.divide(new BigDecimal(1000), 0, RoundingMode.HALF_UP), Units.WATT); // recalculate
-                    // mW to W
-                    logger.trace("New InsightParam standByLimit '{}' for device '{}' received", standByLimit,
+                InsightParser parser = new InsightParser(insightParams);
+                Map<String, State> results = parser.parse();
+                results.forEach((channel, state) -> {
+                    logger.trace("New InsightParam {} '{}' for device '{}' received", channel, state,
                             getThing().getUID());
-                    updateState(WemoBindingConstants.CHANNEL_STANDBYLIMIT, standByLimit);
-
-                    if (currentMW.divide(new BigDecimal(1000), 0, RoundingMode.HALF_UP).intValue() > standByLimitMW
-                            .divide(new BigDecimal(1000), 0, RoundingMode.HALF_UP).intValue()) {
-                        updateState(WemoBindingConstants.CHANNEL_ONSTANDBY, OnOffType.OFF);
-                    } else {
-                        updateState(WemoBindingConstants.CHANNEL_ONSTANDBY, OnOffType.ON);
+                    updateState(channel, state);
+                });
+
+                // Update helper channel onStandBy by checking if currentPower > standByLimit.
+                var standByLimit = (QuantityType<?>) results.get(WemoBindingConstants.CHANNEL_STANDBYLIMIT);
+                if (standByLimit != null) {
+                    var currentPower = (QuantityType<?>) results.get(WemoBindingConstants.CHANNEL_CURRENTPOWER);
+                    if (currentPower != null) {
+                        updateState(WemoBindingConstants.CHANNEL_ONSTANDBY,
+                                currentPower.intValue() > standByLimit.intValue() ? OnOffType.OFF : OnOffType.ON);
```

Comment:
Here, you can rather use `OnOffType(currentPower.intValue() > standByLimit.intValue())`

Reply:
Simplified.